lib/src/main/java/io/ably/lib/util/EventEmitter.java

```diff
@@ -82,8 +82,10 @@ public synchronized void off(Event event, Listener listener) {
 	 * @param args the arguments to pass to listeners
 	 */
 	public synchronized void emit(Event event, Object... args) {
-		for (int i = listeners.size() - 1; i >= 0; i--) {
-			apply(listeners.get(i), event, args);
+		List<Listener> clonedListeners = new ArrayList<>();
```

Comment:
I think here it is worth having a comment with a reference to the spec requirement id `RTE6a`.